enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,435 @@
+
+In the case that a probe was misconfigured, an admin may elect to toggle this behavior off entirely. The ProvisionerClass API, which doesn’t exist today, would be a candidate resource for providing this escape hatch behavior. Without that API in place today, the `plain.core.rukpak.io/enable-probes` BundleDeployment annotation key may serve as a short term solution. The plain provisioner implementation would be responsible for respecting that annotation key.
+
+Injecting this annotation may require changes to the top-level PlatformOperators resource to introduce a toggle field to control this behavior. When the PO manager sees this toggle has been disabled, it's responsible for propagating this configuration as an annotation to the set of BundleDeployment resources it manages.
+
+When reconciling a PO custom resource, the PO manager may need to propagate the underlying BundleDeployment's status back to the PO custom resource that's currently being processed. A separate controller may be responsible for watching, and listing the PO custom resource, and relaying that individual PO status information back to the aggregate ClusterOperator that controller is managing.
+
+### Workflow Description
+
+```mermaid
+sequenceDiagram
+actor Admin
+participant Installer
+participant Bootstrap
+participant CVO
+participant ClusterOperator
+participant PO manager
+Installer->>Admin: Creates the Kubernetes manifests
+Installer->>PO custom resource: Generates
+Admin-->>Installer: Edits Kubernetes manifests
+Installer->>Bootstrap: Creates
+Bootstrap->>CVO: Deploys
+CVO->>PO manager: Deploys
+PO manager->>ClusterOperator: Manages
+PO manager->>PO custom resource: Watches
+CVO->>ClusterOperator: Watches
+```
+
+Note: The above workflow requires admins to manually create, and edit, the cluster Kubernetes manifests with their desired PO intent. There's a chance that admins mistype the package names, and depending on whether upfront validation is possible, that invalid configuration may not be surfaced until after cluster creation. This behavior may require explicit documentation to reduce this risk. These are obvious UX gaps with this workflow, which can be solidified in future phases, but it will result in zero installer changes during the 4.12 timeframe.
+
+### API Extensions
+
+- Adds a top-level PlatformOperator cluster-scoped CRD.
+- Adds the Bundle and BundleDeployment OLM 1.x API(s).
+- Adds a validating webhook for the Bundle and BundleDeployment APIs.
+- Adds a CRD validation webhook.
+
+### Risks and Mitigations
+
+- Early phases will introduce multiple v1alpha1 CRDs downstream.
+- Early phases will introduce OLM v1 components that are in alpha stages.
```

Comment:
the mitigation for these two should be that these new apis are behind a tech preview feature gate

Reply:
Updated in the latest set of commits, and added some more details around the tech preview guidelines.